src/components/movementsensor/MovementSensor.ts

```diff
@@ -0,0 +1,48 @@
+import type { Extra, Orientation, Vector3D } from '../../types';
+
+export interface Position {
+  latitude: number;
+  longitude: number;
+  altitudeMM: number;
+}
+
+export interface Properties {
+  linearVelocitySupported: boolean;
+  angularVelocitySupported: boolean;
+  orientationSupported: boolean;
+  positionSupported: boolean;
+  compassHeadingSupported: boolean;
+  linearAccelerationSupported: boolean;
+}
+
+export interface MovementSensor {
+  /** Get linear velocity in mm/s across x/y/z axes */
+  getLinearVelocity(extra?: Extra): Promise<Vector3D>;
+
+  /** Get the angular velocity in degrees/s across x/y/z axes */
+  getAngularVelocity(extra?: Extra): Promise<Vector3D>;
+
+  /**
+   * Get the compass heading, which is a number from 0-359 where 0 is North, 90
+   * is East, 180 is South, and 270 is West
+   */
+  getCompassHeading(extra?: Extra): Promise<number>;
+
+  /**
+   * Get the compass heading, which is a number from 0-359 where 0 is North, 90
+   * is East, 180 is South, and 270 is West
+   */
+  getOrientation(extra?: Extra): Promise<Orientation>;
+
+  /** Get the current position latitude, longitude, and altitude (in mm) */
+  getPosition(extra?: Extra): Promise<Position>;
+
+  /** Get the movement sensor readings supported by this movement sensor */
```

Comment:
yeah it's a bit unclear - I was trying to say that the "properties" basically tell you what kinds of readings this sensor can return. lemme reword it

Reply:
changed: https://github.com/viamrobotics/viam-typescript-sdk/pull/24/commits/51a4e491fb3fabb205208e243f44470c8c20f915 - let me know if this is better

FYI, in doing this, I noticed that we probably need to export any custom types we define - this is important for both development and for documentation (right now the description I added to the `Properties` interface doesn't get auto-generated).

I'm going to handle this in a separate PR since it might result in a change in the overall export structure of this library.